### PR TITLE
Fix: TreePicker lifecycle

### PR DESCRIPTION
**Pull request by camjc**
- Had issues with mounting and unmounting Treepicker due to async data.
- Handle the async data as suggested at
  https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html
- This is annoying for shallow rendering in tests, so we manually call
  componentDidMount after we render the element to mimic
  what would happen in a normal render.

### CODE REVIEW

File: src/components/adslotUi/TreePickerComponent.js [hunk 1]
```diff
@@ -19,21 +19,32 @@ class TreePickerComponent extends React.Component {
...
 
+    this.state = {};
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
     this.loadData();
   }
 
+  componentWillUnmount() { this._isMounted = false; }
```
Comment (antklim) [inline src/components/adslotUi/TreePickerComponent.js]: Where do we call this method?

**Comment by omgaz** [comment]
Coverage then LGTM

**Comment by vinteo** [comment]
LGTM

File: src/components/adslotUi/TreePickerComponent.js (hunk 1 above)
Comment (camjc) [inline src/components/adslotUi/TreePickerComponent.js]: Yeah, will have to test it.
React calls it, its a lifecycle method.
Our linting guarantees that lifecycle methods will always be at the top of the class btw.

**Comment by vinteo** [comment]
Merging...

**Pull request merged**